dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -0,0 +1,318 @@
+package com.dotmarketing.cms.urlmap;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.business.IdentifierAPI;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.cache.FieldsCache;
+import com.dotmarketing.cms.urlmap.filters.URLMapFilter;
+import com.dotmarketing.common.model.ContentletSearch;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.structure.StructureUtil;
+import com.dotmarketing.portlets.structure.factories.StructureFactory;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.dotmarketing.portlets.structure.model.SimpleStructureURLMap;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.RegExMatch;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.util.StringPool;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.TreeSet;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.jetbrains.annotations.NotNull;
+
+public class URLMapAPIImpl {
+
+    private Collection<PatternCache> patternsCache;
+
+    private final UserWebAPI wuserAPI = WebAPILocator.getUserWebAPI();
+    private final HostWebAPI whostAPI = WebAPILocator.getHostWebAPI();
+    private final ContentletAPI contentletAPI = APILocator.getContentletAPI();;
+    private final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+    private final IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
+
+    public URLMapInfo processURLMap(final UrlMapContext context)
```

Comment:
probably returning an optional URLMapInfo could be more fashioned.

Reply:
Agree